File: easy/3487_maximum_unique_subarray_sum_after_deletion/solution.py
from typing import List


class Solution:
    def maxSum(self, nums: List[int]) -> int:
        unique_positive_nums = [num for num in set(nums) if num > 0]

        if len(unique_positive_nums) == 0:
            return max(nums)

        return sum(unique_positive_nums)

File: easy/3487_maximum_unique_subarray_sum_after_deletion/solution.cpp
using namespace std;

#include <algorithm>
#include <numeric>
#include <unordered_set>
#include <vector>

class Solution {
public:
  int maxSum(vector<int> &nums) {
    unordered_set<int> uniqueNums(nums.begin(), nums.end());

    vector<int> uniquePositiveNums;
    copy_if(uniqueNums.begin(), uniqueNums.end(),
            back_inserter(uniquePositiveNums), [](int num) { return num > 0; });

    if (uniquePositiveNums.size() == 0)
      return *max_element(nums.begin(), nums.end());

    return accumulate(uniquePositiveNums.begin(), uniquePositiveNums.end(), 0);
  }
};
